local_modules/web3-providers-http/local_modules/xhr2-cookies/xml-http-request.ts

```diff
@@ -0,0 +1,471 @@
+	anon?: boolean;
+}
+export interface XHRUrl extends Url {
+	method?: string;
+}
+
+export class XMLHttpRequest extends XMLHttpRequestEventTarget {
+	static ProgressEvent = ProgressEvent;
+	static InvalidStateError = InvalidStateError;
+	static NetworkError = NetworkError;
+	static SecurityError = SecurityError;
+	static SyntaxError = SyntaxError;
+	static XMLHttpRequestUpload = XMLHttpRequestUpload;
+
+	static UNSENT = 0;
+	static OPENED = 1;
+	static HEADERS_RECEIVED = 2;
+	static LOADING = 3;
+	static DONE = 4;
+
+	static cookieJar = Cookie.CookieJar();
+
+	UNSENT = XMLHttpRequest.UNSENT;
+	OPENED = XMLHttpRequest.OPENED;
+	HEADERS_RECEIVED = XMLHttpRequest.HEADERS_RECEIVED;
+	LOADING = XMLHttpRequest.LOADING;
+	DONE = XMLHttpRequest.DONE;
+
+	onreadystatechange: ProgressEventListener | null = null;
+	readyState: number = XMLHttpRequest.UNSENT;
+
+	response: string | ArrayBuffer | Buffer | object | null = null;
+	responseText = '';
+	responseType = '';
+	status = 0; // TODO: UNSENT?
+	statusText = '';
+	timeout = 0;
+	upload = new XMLHttpRequestUpload();
+	responseUrl = '';
+	withCredentials = false;
+
+	nodejsHttpAgent: HttpsAgent;
+	nodejsHttpsAgent: HttpsAgent;
+	nodejsBaseUrl: string | null;
+
+	private _anonymous: boolean;
+	private _method: string | null = null;
+	private _url: XHRUrl | null = null;
+	private _sync = false;
+	private _headers: {[header: string]: string} = {};
+	private _loweredHeaders: {[lowercaseHeader: string]: string} = {};
+	private _mimeOverride: string | null = null; // TODO: is type right?
+	private _request: ClientRequest | null = null;
+	private _response: IncomingMessage | null = null;
+	private _responseParts: Buffer[] | null = null;
+	private _responseHeaders: {[lowercaseHeader: string]: string} | null = null;
+	private _aborting = null; // TODO: type?
+	private _error = null; // TODO: type?
+	private _loadedBytes = 0;
+	private _totalBytes = 0;
+	private _lengthComputable = false;
+
+	private _restrictedMethods = {CONNECT: true, TRACE: true, TRACK: true};
+	private _restrictedHeaders = {
+		'accept-charset': true,
+		'accept-encoding': true,
+		'access-control-request-headers': true,
+		'access-control-request-method': true,
+		connection: true,
+		'content-length': true,
+		cookie: true,
+		cookie2: true,
+		date: true,
+		dnt: true,
+		expect: true,
+		host: true,
+		'keep-alive': true,
+		origin: true,
+		referer: true,
+		te: true,
+		trailer: true,
+		'transfer-encoding': true,
+		upgrade: true,
+		'user-agent': true,
+		via: true
+	};
+	private _privateHeaders = {'set-cookie': true, 'set-cookie2': true};
+
+	//Redacted private information (${os.type()} ${os.arch()}) node.js/${process.versions.node} v8/${process.versions.v8} from the original version @ github
+	//Pretend to be tor-browser https://blog.torproject.org/browser-fingerprinting-introduction-and-challenges-ahead/
+	private _userAgent = `Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0`;
```

Comment:
This will hide User-Agent exposing the version of node.js and os to protect privacy when torPort is enabled. Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0 is the User-Agent of the latest version of Tor Browser.